### PR TITLE
Add script to automate cutting a release

### PR DESCRIPTION
**Pull request by marcofranssen**
Run `./release-chart.sh --help` for usage information.

This is how the PR description will look like e.g.:

```shell
./release-chart.sh --chart spire --current-version 0.3.0 --new-version 0.4.0 --dry-run
```

Results in 👇 

----------------------------------------

Please review the below changelog to ensure this matches up with the semantic version being applied.

> **Note**: **Maintainers** ensure to run following after merging this PR to trigger the release workflow:
>
> ```shell
> git checkout main
> git pull
> git checkout release
> git pull
> git merge main
> git push
> ```

**Changes in this release**

* eb186ca Add values for server TTL configurables
* f123296 Add a flag to configure the Kubernetes NodeAttestor (#83)
* 0275569 Incorperate feedback
* a00c97b Incorperate feedback
* f93434a Make server service account allow list configurable
* f0b7f5b Make spire agent server address configurable
* cb627f0 Allow all subchart to conditionally disable (#90)
* 2e9510f Switch to non root nginx (#89)
* a901751 Fix Homepage URL
* 06c395f Fix sources in Chart.yaml (#85)
* d341c5a UpstreamAuthority cert-manager support (#82)
* a82ee69 Make sure all configmap changes rollout
* b7f8c86 Add extra initContainers, containers, volumes to agent and server
* 81ac89a Add k8s 1.26.0 (Kind) to the test matrix
* bb6abfc Merge pull request #76 from spiffe/dependabot/github_actions/sigstore/cosign-installer-3.0.1
* c0e5665 Bump cosign to v2.0.0
* d729a44 Bump sigstore/cosign-installer from 2.8.1 to 3.0.1
* fe7cd73 Merge pull request #74 from spiffe/arm64-support
* a3c04c0 Update supported versions
* 75480ed Bump spiffe-csi-driver image to 0.2.3 (arm64 support)
* 79889ff Bump spiffe-oidc-discovery-provider image to 1.6.0 (arm64 support)
* a632f76 Bump spire-agent image to 1.6.0 (arm64 support)
* b92e81d Bump spire-server image to 1.6.0 (arm64 support)
* fea2aa5 Fix sigstore/cosign-installer usage in release workflow
* fe93346 Use quote function instead of actual quotes
* 9480ab7 Add beta note to readme (#62)
* 4681498 Better spire-server entry commands (#59)

### CODE REVIEW

**Review by kfox1111 — APPROVED** [review]
lgtm

File: release-chart.sh [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+commits_since_previous_release="$(git log "${chart}-${current_version}..HEAD" --pretty=format:'* %h %s')"
+"${SED}" -i "s/version: ${current_version}/version: ${new_version}/" "charts/${chart}/Chart.yaml"
+./helm-docs.sh
+git add "charts/${chart}/"{Chart.yaml,README.md}
+git commit -m "Bump ${chart} Helm Chart version from ${current_version} to ${new_version}" \
+  -m "${commits_since_previous_release}" \
+  -s
+git push -u --force
+
+cat <<EOF | gh pr create --base main --body-file -
```
Comment (marcofranssen) [inline release-chart.sh:110]:
```suggestion
cat <<EOF | gh pr create --base main --body-file - -w
```

**Comment by faisal-memon** [comment]
Output when i run it:

```
./release-chart.sh --chart spire --current-version 0.3.0 --new-version 0.4.0 --dry-run

Switched to branch 'main'
Your branch is up to date with 'origin/main'.
Already up to date.
Switched to and reset branch 'bump-spire-version'
Using 'helm-docs version 1.11.0'
INFO[2023-03-07T19:20:41-08:00] Found Chart directories [charts/spire, charts/spire/charts/spiffe-csi-driver, charts/spire/charts/spiffe-oidc-discovery-provider, charts/spire/charts/spire-agent, charts/spire/charts/spire-server]
INFO[2023-03-07T19:20:41-08:00] Generating README Documentation for chart charts/spire/charts/spire-agent
INFO[2023-03-07T19:20:41-08:00] Generating README Documentation for chart charts/spire/charts/spiffe-oidc-discovery-provider
INFO[2023-03-07T19:20:41-08:00] Generating README Documentation for chart charts/spire/charts/spiffe-csi-driver
INFO[2023-03-07T19:20:41-08:00] Generating README Documentation for chart charts/spire
INFO[2023-03-07T19:20:41-08:00] Generating README Documentation for chart charts/spire/charts/spire-server
diff --git a/charts/spire/Chart.yaml b/charts/spire/Chart.yaml
index 00a0705..ef7008a 100644
--- a/charts/spire/Chart.yaml
+++ b/charts/spire/Chart.yaml
@@ -3,7 +3,7 @@ name: spire
 description: >
   A Helm chart for deploying the complete Spire stack including: spire-server, spire-agent, spiffe-csi-driver, spiffe-oidc-discovery-provider and spire-controller-manager.
 type: application
-version: 0.3.0
+version: 0.4.0
 appVersion: "1.5.5"
 keywords: ["spiffe", "spire", "spire-server", "spire-agent", "oidc", "spire-controller-manager"]
 home: https://github.com/spiffe/helm-charts/tree/main/charts/spire
diff --git a/charts/spire/README.md b/charts/spire/README.md
index b99a99c..78d0d1d 100644
--- a/charts/spire/README.md
+++ b/charts/spire/README.md
@@ -2,7 +2,7 @@

 <!-- This README.md is generated. Please edit README.md.gotmpl -->

-![Version: 0.3.0](https://img.shields.io/badge/Version-0.3.0-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 1.5.5](https://img.shields.io/badge/AppVersion-1.5.5-informational?style=flat-square)
+![Version: 0.4.0](https://img.shields.io/badge/Version-0.4.0-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 1.5.5](https://img.shields.io/badge/AppVersion-1.5.5-informational?style=flat-square)

 A Helm chart for deploying the complete Spire stack including: spire-server, spire-agent, spiffe-csi-driver, spiffe-oidc-discovery-provider and spire-controller-manager.

[bump-spire-version 1d953ce] Bump spire Helm Chart version from 0.3.0 to 0.4.0
 2 files changed, 2 insertions(+), 2 deletions(-)
fatal: The current branch bump-spire-version has no upstream branch.
To push the current branch and set the remote as upstream, use

    git push --set-upstream origin bump-spire-version

To have this happen automatically for branches without a tracking
upstream, see 'push.autoSetupRemote' in 'git help config'.

no pull requests found for branch "bump-spire-version"
Switched to branch 'main'
Your branch is up to date with 'origin/main'.
```

**Comment by marcofranssen** [comment]
@faisal-memon I added some more explicit options which might be in my global `.gitconfig`. Could you try again?

**Pull request merged**